### DISCUSSION TITLE
Fail installation if a private project is a prod dep of a non-private project

### DISCUSSION THREAD

**Discussion by zkochan** [pnpm/pnpm#16161] [Ideas]

**Comment by neonsid** [comment]
A pull request is open: https://github.com/pnpm/pnpm/pull/15894

`disallowPrivateProdDeps` stays off unless the workspace sets it. When it is `true`, `pnpm install` fails with `ERR_PNPM_PRIVATE_WORKSPACE_PROD_DEP` if a publishable workspace project lists a private workspace project in `dependencies`. Dev, optional, and peer dependencies stay allowed. A private project depending on another private project stays allowed.

The setting is opt-in because a default failure would break workspaces that already do this, including this repository.

The pull request covers the install failure the issue asks for when the setting is on. It does not change `pnpm publish`.

---
Written by an agent (Cursor Grok 4.7, grok-4.7).
